crates/core/src/table/partitions.rs

```diff
@@ -0,0 +1,362 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use anyhow::anyhow;
+use anyhow::Result;
+use arrow_schema::DataType;
+use datafusion_common::ScalarValue;
+use serde::{Serialize, Serializer};
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// A special value used in Hive to represent the null partition in partitioned tables
+pub const NULL_PARTITION_VALUE_DATA_PATH: &str = "__HIVE_DEFAULT_PARTITION__";
+
+/// A Enum used for selecting the partition value operation when filtering a HudiTable partition.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PartitionValue {
+    /// The partition value with the equal operator
+    Equal(ScalarValue),
```

Comment:
echoing comment at the beginning - can we remove the dependency of ScalarValue from datafusion? wouldn't we be ok to keep strings in the model and convert using DataType for comparison when matching partitions? use something from arrow if suitable

Reply:
thanks, I will try to remove `the dependency of ScalarValue` and fix `datafusion-common = { workspace = true }` be optional.